reports/src/main/java/com/ivy/reports/ReportViewModel.kt

```diff
@@ -1,415 +1,418 @@
+    private val queryTransactionsAct: TrnsAct,
+    private val calculateAct: CalculateWithTransfersAct,
+    private val exchangeAct: ExchangeAct,
+    private val baseCurrencyAct: BaseCurrencyAct,
+    private val groupTransactionsAct: GroupTrnsAct,
     private val exportCSVLogic: ExportCSVLogic,
-    private val exchangeAct: ExchangeActOld,
-    private val accountsAct: AccountsActOld,
-    private val categoriesAct: CategoriesActOld,
-    private val trnsWithDateDivsAct: TrnsWithDateDivsAct,
-    private val calcTrnsIncomeExpenseAct: CalcTrnsIncomeExpenseAct,
-    private val baseCurrencyAct: BaseCurrencyActOld
-) : FRPViewModel<ReportScreenState, Nothing>() {
+    private val plannedPaymentsLogic: PlannedPaymentsLogic,
+) : FRPViewModel<ReportScreenState, ReportScreenEvent>() {
     override val _state: MutableStateFlow<ReportScreenState> = MutableStateFlow(
         ReportScreenState()
     )
 
-    override suspend fun handleEvent(event: Nothing): suspend () -> ReportScreenState {
-        TODO("Not yet implemented")
-    }
-
-    private val unSpecifiedCategory =
-        CategoryOld(com.ivy.core.ui.temp.stringRes(R.string.unspecified), color = Gray.toArgb())
-
-    private val _period = MutableLiveData<TimePeriod>()
-    val period = _period.asLiveData()
-
-    private val _categories = MutableStateFlow<List<CategoryOld>>(emptyList())
-    val categories = _categories.readOnly()
-
-    private val _allAccounts = MutableStateFlow<List<AccountOld>>(emptyList())
-
     private val _baseCurrency = MutableStateFlow("")
-    val baseCurrency = _baseCurrency.readOnly()
-
-    private val _historyIncomeExpense = MutableStateFlow(IncomeExpenseTransferPair.zero())
-    private val historyIncomeExpense = _historyIncomeExpense.readOnly()
-
     private val _filter = MutableStateFlow<ReportFilter?>(null)
-    val filter = _filter.readOnly()
-
-    fun start() {
-        viewModelScope.launch(Dispatchers.IO) {
-            _baseCurrency.value = baseCurrencyAct(Unit)
-            _allAccounts.value = accountsAct(Unit)
-            _categories.value = listOf(unSpecifiedCategory) + categoriesAct(Unit)
+    private val allAcc = MutableStateFlow<List<Account>>(emptyList())
+    private val allCategories = MutableStateFlow<List<Category>>(emptyList())
+    private val transStatsGlobal = MutableStateFlow(ExtendedStats.empty())
+    private val allTransactions = MutableStateFlow<List<Transaction>>(emptyList())
+    private val categoryNone = dummyCategory(
+        name = "None",
+        color = Gray.toArgb(),
+        icon = dummyIconSized(R.drawable.ic_custom_category_s)
+    )
 
-            updateState {
-                it.copy(
-                    baseCurrency = _baseCurrency.value,
-                    categories = _categories.value,
-                    accounts = _allAccounts.value
-                )
+    override suspend fun handleEvent(event: ReportScreenEvent): suspend () -> ReportScreenState =
+        withContext(Dispatchers.Default) {
```

Comment:
Why do we need that? Ivy FRP handles threading out-of-the-box?

Reply:
Ivy FRP handles threading indeed handles the threading by default..
 
By default FRP is launched via the following code
viewModelScope.launch {
//Code 
}

Since there is no option to specify the threadpool it runs on the code runs on the UI thread by default

Reply:
Ohh, my bad! Actions handle threading by default in IO. Leave it like that 👍 

IMO, any computation heavy work should be done via actions.